config/initializers/014-track-setting-changes.rb

```diff
@@ -34,6 +34,8 @@
 
   Jobs.enqueue(:update_s3_inventory) if [:s3_inventory, :s3_upload_bucket].include?(name)
 
+  Jobs.enqueue(:update_private_uploads_acl) if [:prevent_anons_from_downloading_files].include?(name)
```

Comment:
```suggestion
  Jobs.enqueue(:update_private_uploads_acl) if name == :prevent_anons_from_downloading_files
```
:) 

One interesting general problem we have is if this setting is shadowed by global this thing will never trigger.